### ISSUE TITLE
Small Mirror issue

### ISSUE THREAD

**Issue by seularts** [jayanam/fast-carve#3]
I love your addon. It's simple and elegant and works right out of the box. I did find a small issue though when using the move and rotate options. For example: when I create a new object that will be subtracted from the main object it will span in the center of the first object (where the cursor is usually set). Now if I press G and move it towards the X-axis, I will want to rotate it on the top view 45 degrees relative to the center of the main object. I use R and set the Cursor to be the center reference of the rotation while pressing Ctrl to go in increments. All nice and easy up to this point but now when I apply the Center origin and mirror, it mirrors my object diagonally to the axis instead of side by side. An easy fix to this is to go into the mirror modifier and set up the main object in the Mirror Object field at the bottom. I believe this is something that can be solved relatively easy in your code as it just doesn't recognize the center origin for some reason after it has been rotated.

**Comment by jayanam** [comment]
Hi,
try to apply the rotation of the object before you add the mirror modifier,
if you rotated the object in object mode. If you rotated in edit mode, you
dont need to apply.

Cheers
Jay

**Comment by seularts** [comment]
I did do the rotation first. The mirror modifier acts weird after I place the object using the move and rotate.

**Comment by jayanam** [comment]
Yes, this is default behavior. When you transform in object mode you have to apply the transform. Scale is another example: Try to scale an object in object mode, then add some bevels to edges and you see that the results look strange cause transform in object mode does not transform your mesh.

**Comment by seularts** [comment]
Ah, I see. Awesome, thanks for the replies :)

**Comment by jayanam** [comment]
You are welcome, I have an older video about this issue if you are
interested:
https://www.youtube.com/watch?v=MYYO1JbC7b8
